src/StoreApi/Formatters.php

```diff
@@ -0,0 +1,45 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\StoreApi;
+
+use Exception;
+use Schemas\AbstractSchema;
+use Automattic\WooCommerce\Blocks\Domain\Services\ExtendRestApi;
+use Automattic\WooCommerce\Blocks\StoreApi\Formatters\AbstractFormatter;
+use Automattic\WooCommerce\Blocks\StoreApi\Formatters\DefaultFormatter;
+
+/**
+ * Formatters class.
+ *
+ * Allows formatter classes to be registered. Formatters are exposed to extensions via the ExtendRestApi class.
+ */
+class Formatters {
+	/**
+	 * Holds an array of formatter classes.
+	 *
+	 * @var array
+	 */
+	private $formatters = [];
+
+	/**
+	 * Get a new instance of a formatter class.
+	 *
+	 * @param string $name Name of the formatter.
+	 * @return AbstractFormatter Formatter class instance.
+	 */
+	public function __get( $name ) {
+		if ( ! isset( $this->formatters[ $name ] ) ) {
+			return new DefaultFormatter();
+		}
+		return new $this->formatters[ $name ]();
+	}
```

Comment:
Does this mean we're going to just die silently when someone reaches an unknown formatter? 
Can't we do this for everything but throw an error for admin users?

Reply:
This doesn't die silently. This will return a default formatter which passes through values. This avoids extensions shooting themselves in the foot by using unregistered formatters, or breakages if a formatter is removed or renamed.

Reply:
This is my definition of dying silenctly (which is wrong).
I think the current behaviour is great and that's what I'm doing everywhen when it makes sense, I was suggesting on top of that we also throw an error for admin users only.
Same we do in extendRestApi
```
private function throw_exception( $exception_or_error ) {
		if ( is_string( $exception_or_error ) ) {
			$exception = new Exception( $exception_or_error );
		} else {
			$exception = $exception_or_error;
		}
		// Always log an error.
		wc_caught_exception( $exception );
		if ( defined( 'WP_DEBUG' ) && WP_DEBUG && current_user_can( 'manage_woocommerce' ) ) {
			throw $exception;
		}
	}
```